independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanProcessor.java

```diff
@@ -240,6 +243,9 @@ private static IndexView addBuiltinClasses(IndexView index) {
             index(indexer, Default.class.getName());
             index(indexer, Any.class.getName());
             index(indexer, Named.class.getName());
+            index(indexer, Initialized.class.getName());
+            index(indexer, BeforeDestroyed.class.getName());
+            index(indexer, Destroyed.class.getName());
```

Comment:
I suppose you have the full list now but I wonder if it would be more future proof to index the full API jar instead of specifying each element manually?

Reply:
Yes, you're right. This is just a temporary fix. We need a better solution because CDI API is not the only jar that could contain qualifiers.